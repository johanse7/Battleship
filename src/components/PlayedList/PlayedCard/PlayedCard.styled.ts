import styled from "styled-components"
import { displays } from "../../../theme/base/Variables"

const PlayedCardStyled = styled.article<{ win: boolean }>`
  background: ${({ theme }) => theme.palette.common.black};
  border: 2px solid
    ${({ theme, win }) => (win ? theme.palette.quaternary.main : theme.palette.tertiary.main)};

  color: ${({ theme }) => theme.palette.common.grey20};
  ${displays.flexColumn};
  align-items: stretch;
  gap: 1rem;

  & .header {
    ${displays.flexCenter};
    font-size: 1.2rem;
    gap: 1rem;
    & > svg {
      font-size: 3rem;
      color: ${({ theme, win }) =>
        win ? theme.palette.primary.contrastText : theme.palette.tertiary.main};
      font-weight: bold;
    }
  }
  & .body {
    ${displays.flexBase};
    gap: 1rem;

    & .turns {
      font-size: 2rem;
      color: ${({ theme, win }) =>
        win ? theme.palette.primary.contrastText : theme.palette.tertiary.main};
      font-weight: bold;
    }
  }

  padding: 1rem; ;
`

export default PlayedCardStyled
