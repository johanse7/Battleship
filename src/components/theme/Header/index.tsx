import { useLocation } from "react-router-dom"
import { Battleship, IconGame, IconSettings } from "../../../theme/common/Icons.styled"
import { Linktyled } from "../../../theme/common/Link"
import HeaderStyled, { HeaderSettingsStyled } from "./Header.styled"

const Header = () => {
  const location = useLocation()

  return (
    <HeaderStyled>
      <div className="title">
        <p>Battleship</p>
        <Battleship />
      </div>

      {location.pathname === "/" && (
        <Linktyled to="/playedlist">
          <HeaderSettingsStyled>
            <IconGame />
            <span>Played games</span>
          </HeaderSettingsStyled>
        </Linktyled>
      )}

      {(location.pathname.includes("board") || location.pathname.includes("playedlist")) && (
        <Linktyled to="/">
          <HeaderSettingsStyled>
            <IconSettings />
            <span>Settings</span>
          </HeaderSettingsStyled>
        </Linktyled>
      )}
    </HeaderStyled>
  )
}

export default Header
