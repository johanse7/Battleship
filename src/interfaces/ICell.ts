export interface ICell {
  positionX: number
  positionY: number
  isSelected: boolean
  hasShip: boolean
}

export interface ICellExtend extends ICell {
  onClickCell: () => void
}
