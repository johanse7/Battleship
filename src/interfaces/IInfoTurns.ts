export interface IInfoTurns {
  turnUsed: number
  ships: number
}
