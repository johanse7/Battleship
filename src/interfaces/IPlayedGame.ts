export interface IPlayedGame {
  win: boolean
  turns: number
  levelName: string
}
