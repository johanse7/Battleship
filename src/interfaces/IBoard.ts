import { ICell } from "./ICell"

export interface IBoard {
  matrix: Array<Array<ICell>>
  finishedGame: boolean
  onClickCell: (cell: ICell) => void
}

export interface IBoardConfig {
  size: number
  maxLengthShips: number
}

export interface IBoardState {
  board: Array<Array<ICell>>
  turnsUsed: number
  shootDown: number
  countBlockShip: number
  ships: number
  resultGame: {
    win: boolean
  } | null
}
