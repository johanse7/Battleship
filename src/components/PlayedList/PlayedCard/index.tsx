import { IPlayedGame } from "../../../interfaces/IPlayedGame"
import { IconCheck, IconError } from "../../../theme/common/Icons.styled"
import PlayedCardStyled from "./PlayedCard.styled"

const PlayedCard = ({ levelName, turns, win }: IPlayedGame) => {
  return (
    <PlayedCardStyled win={win}>
      <div className="header">
        <p>{levelName}</p>
        {win ? <IconCheck /> : <IconError />}
      </div>
      <div className="body">
        <span>Turns used: </span>
        <span className="turns">{turns}!!</span>
      </div>
    </PlayedCardStyled>
  )
}

export default PlayedCard
