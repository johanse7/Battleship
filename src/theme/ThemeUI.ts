import { ITheme } from "../interfaces/ITheme"
import { greyScale, fontSizes } from "./base/Variables"

export const defaultTheme: ITheme = {
  palette: {
    common: {
      black: "#101010",
      white: "#ffffff",
      ...greyScale,
    },
    primary: {
      main: "#1EF400",
      contrastText: "#108800",
    },
    secondary: {
      main: "#709fb0",
      contrastText: "rgb(20,90,4)",
    },
    tertiary: {
      main: "#4F1005",
    },
    quaternary: {
      main: "#FFD700",
    },
  },
  button: {
    primary: {
      text: greyScale.grey20,
      background: "#101010",
    },
    secondary: {
      background: greyScale.grey80,
      text: greyScale.grey20,
    },
  },

  fontSizes,
}
