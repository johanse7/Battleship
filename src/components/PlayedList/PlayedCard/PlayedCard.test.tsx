import "@testing-library/jest-dom/extend-expect"
import "jest-styled-components"
import { render } from "@testing-library/react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import { defaultTheme } from "../../../theme/ThemeUI"
import { IPlayedGame } from "../../../interfaces/IPlayedGame"
import PlayedCard from "./"

test("render icon played card ", () => {
  const card: IPlayedGame = {
    levelName: "Easy level",
    turns: 20,
    win: true,
  }

  const component = render(
    <ThemeProvider theme={defaultTheme}>
      <PlayedCard {...card} />
    </ThemeProvider>,
  )

  component.container.querySelector("svg")
})
