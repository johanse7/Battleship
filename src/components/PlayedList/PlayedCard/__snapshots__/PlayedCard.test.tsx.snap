// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
.c0 {
  background: #101010;
  border: 2px solid #FFD700;
  color: #CBCCD8;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  gap: 1rem;
  padding: 1rem;
}

.c0 .header {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.2rem;
  gap: 1rem;
}

.c0 .header > svg {
  font-size: 3rem;
  color: #108800;
  font-weight: bold;
}

.c0 .body {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1rem;
}

.c0 .body .turns {
  font-size: 2rem;
  color: #108800;
  font-weight: bold;
}

<article
  className="c0"
/>
`;
